jobbergate-api/alembic/versions/20240226_175811--35ccf022687d_replace_slurmrestd_by_sbatch.py

```diff
@@ -0,0 +1,41 @@
+"""Replace slurmrestd by sbatch
+
+Revision ID: 35ccf022687d
+Revises: ec2d2948fb41
+Create Date: 2024-02-26 17:58:11.147803
+
+"""
+
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "35ccf022687d"
+down_revision = "ec2d2948fb41"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column("job_submissions", sa.Column("sbatch_arguments", sa.String(), nullable=True))
```

Comment:
what do you think about this column being a JSONB column instead of a string? To me, it makes much more sense to parse the batch arguments instead of just putting them as it comes.

Reply:
Now, to your point. I've considered keeping it as a JSONB, having a parser may result in early error feedback to the users, and a few other points. But at the same time, it would impose constrains on the sbatch instances and versions the system as a whole can operate with.

One of the main goals in replacing slurmrestd by sbatch is:
> Extra functionality can be added to the sbatch command by installing plugins. For instance, we can cite the support to run containerized tasks added by pyxis. Such flexibility is not possible by using Slurm Rest API.

So to make the API more flexible and considering it just stores this data, I think it should be up to slurm on each cluster to decide it they are fine with the arguments or not (reject the job in this case). It is even more important when we consider the multi-tenancy scenario, where different clusters can operate on different settings and sbatch versions, and the API have to support them all.

Reply:
So, `--name=foo -c bar --flag` should be stored as the follow?
```json
{
  "name": "foo",
  "c": "bar",
  "flag": null,
}
```

Reply:
It could be, yes. However, I think we need to consider how these params would be passed in to the API first. Would Jobbergate CLI just ask the user to pass in a string of sbatch params? It would be very clunky in the API to have the user just supply a blob of sbatch params in a single string, I think.

Reply:
We agreed that a list of strings is the choice here. It was addressed in [a16b72f](https://github.com/omnivector-solutions/jobbergate/pull/497/commits/a16b72f5a02d903d4bfc5a01210f83578bff66cc).